src/GuppyDataExplorer/ExplorerTable/index.jsx

```diff
@@ -248,10 +252,46 @@ class ExplorerTable extends React.Component {
     this.setState({ showEmptyColumns: checked });
   };
 
+  /**
+   * processes the table data if this is a dicom table. This will determine if the table row
+   * has image data and will then esure that the lick icon is shown
+   */
+  augmentData = () => {
+    const haveField = this.props.rawData.filter((x) => Object.keys(x).includes(this.props.tableConfig.dicomViewerId));
+    if (haveField.length === this.props.rawData.length) {
+      this.setState({ loading: true });
+      // eslint-disable-next-line array-callback-return
+      Promise.all(this.props.rawData.map((x) => {
+        const dicomServerLink = `${hostname}dicom-server/dicom-web/studies/${x[this.props.tableConfig.dicomViewerId]}/series`;
+        return fetch(dicomServerLink, {
+          method: 'GET',
+        })
+          .then((resp) => ({ ...x, has_dicom_images: resp.headers.get('content-length') !== '22' }),
```

Comment:
what if the request fails here?

Reply:
The idea is to check if it has any response, so if it fails it means that the image is not available, and then we won't show the button.

Reply:
I guess my question is why it is checking for content length? does that endpoint returns 200 even if the DICOM viewer ID doesn;t exists?

Reply:
Yes, this endpoint will return 200 if there is no series available.

To be more specific, it would be an empty list:
```
[]
```

Which has `Content-Length: 22`.

Reply:
thanks for the explanation